### PR TITLE
[release-4.13] OCPBUGS-38254: [CARRY] perform operator apiService certificate validity checks directly

### PR DESCRIPTION
**Pull request by ankitathomas**
Manual cherry-pick of https://github.com/openshift/operator-framework-olm/pull/821 to 4.14

### CODE REVIEW

**Comment by jianzhangbjz** [comment]
Hi @ankitathomas , I got the below error when building cluster with this PR, could you help have a look when you get a chance? Thanks! https://prow.ci.openshift.org/view/gs/test-platform-results/logs/release-openshift-origin-installer-launch-gcp-modern/1822827253600358400 
```console
Go compliance shim [6124] [rhel-8-golang-1.19][openshift-golang-builder]: invoking real go binary
# github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install
vendor/github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install/certresources.go:280:19: undefined: sets.New
Go compliance shim [6124] [rhel-8-golang-1.19][openshift-golang-builder]: Exited with: 1
make[1]: Leaving directory '/build'
make[1]: *** [Makefile:79: github.com/operator-framework/operator-lifecycle-manager/cmd/catalog] Error 1
make: *** [Makefile:67: build/olm] Error 2
error: build error: building at STEP "RUN make build/olm bin/cpb": while running runtime: exit status 2
```

**Comment by jianzhangbjz** [comment]
Test pass, details: https://issues.redhat.com/browse/OCPBUGS-38254
/lgtm
/label qe-approved
/label cherry-pick-approved

**Comment by jianzhangbjz** [comment]
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_operator-framework-olm/836/pull-ci-openshift-operator-framework-olm-release-4.13-e2e-gcp-olm/1824389101466423296
```console
level=error msg=Cluster operator kube-scheduler Degraded is True with MissingStaticPodController_SyncError::StaticPods_Error: MissingStaticPodControllerDegraded: static pod lifecycle failure - static pod: "openshift-kube-scheduler" in namespace: "openshift-kube-scheduler" for revision: 7 on node: "ci-op-di2mvmr9-ae41b-g86vb-master-1" didn't show up, waited: 3m0s
level=error msg=StaticPodsDegraded: pod/openshift-kube-scheduler-ci-op-di2mvmr9-ae41b-g86vb-master-1 container "kube-scheduler" is terminated: Completed: 
level=error msg=StaticPodsDegraded: pod/openshift-kube-scheduler-ci-op-di2mvmr9-ae41b-g86vb-master-1 container "kube-scheduler-cert-syncer" is terminated: Error: go:169: Failed to watch *v1.Secret: failed to list *v1.Secret: Get "https://localhost:6443/api/v1/namespaces/openshift-kube-scheduler/secrets?limit=500&resourceVersion=0": x509: certificate signed by unknown authority
level=error msg=StaticPodsDegraded: W0816 10:51:46.493141       1 reflector.go:424] k8s.io/client-go@v0.26.10/tools/cache/reflector.go:169: failed to list *v1.ConfigMap: Get "https://localhost:6443/api/v1/namespaces/openshift-kube-scheduler/configmaps?limit=500&resourceVersion=0": x509: certificate signed by unknown authority
...
```

**Comment by perdasilva** [comment]
@ankitathomas e2e is failing because of another fetchCSV with the wrong ordering in the parameters (namespace <-> name). The flake (blocks a CRD upgrade that could cause data loss) is a perma fail. While I think the workload is still being protected (i.e. the upgrade doesn't go through), it seems the IP isn't getting updated with the error (at least not in the right place) ? Or being put in the right state after detecting it?

File: staging/operator-lifecycle-manager/test/e2e/csv_e2e_test.go [hunk 1]
```diff
@@ -3673,27 +3672,40 @@ var _ = Describe("ClusterServiceVersion", func() {
...
-			fetchedCSV, err := fetchCSV(crc, csv.Name, generatedNamespace.GetName(), csvSucceededChecker)
+			fetchedCSV, err := fetchCSV(crc, generatedNamespace.GetName(), csv.Name, csvSucceededChecker)
 			Expect(err).ShouldNot(HaveOccurred())
 
 			// Update csv with modified deployment spec
 			fetchedCSV.Spec.InstallStrategy.StrategySpec = strategyNew
 
 			Eventually(func() error {
+				// Fetch the current csv
+				fetchedCSV, err := fetchCSV(crc, csv.Name, generatedNamespace.GetName(), csvSucceededChecker)
```
Comment (perdasilva) [inline staging/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:3683]:
```suggestion
				fetchedCSV, err := fetchCSV(crc, generatedNamespace.GetName(), csv.Name, csvSucceededChecker)
```

**Comment by jianzhangbjz** [comment]
It failed at:
```console
Summarizing 1 Failure:
  [FAIL] CRD Versions [It] [FLAKE] blocks a CRD upgrade that could cause data loss
  /go/src/github.com/openshift/operator-framework-olm/staging/operator-lifecycle-manager/test/e2e/crd_e2e_test.go:275
Ran 8 of 199 Specs in 305.024 seconds
FAIL! -- 7 Passed | 1 Failed | 2 Pending | 189 Skipped
--- FAIL: TestEndToEnd (305.04s)
```

**Comment by tmshort** [comment]
That is a known flake.
/lgtm

**Comment by jianzhangbjz** [comment]
Hi @tmshort , I guess it needs the `backport-risk-assessed` label.

**Pull request merged**